src/tiled/propertybrowser.cpp

```diff
@@ -857,6 +833,47 @@ QtVariantProperty *PropertyBrowser::createProperty(PropertyId id, int type,
     return property;
 }
 
+void PropertyBrowser::addProperties()
+{
+    mUpdating = true;
+
+    // Add the built-in properties for each object type
+    switch (mObject->typeId()) {
+    case Object::MapType:               addMapProperties(); break;
+    case Object::MapObjectType:         addMapObjectProperties(); break;
+    case Object::LayerType:
+        switch (static_cast<Layer*>(mObject)->layerType()) {
+        case Layer::TileLayerType:      addTileLayerProperties();   break;
+        case Layer::ObjectGroupType:    addObjectGroupProperties(); break;
+        case Layer::ImageLayerType:     addImageLayerProperties();  break;
+        }
+        break;
+    case Object::TilesetType:           addTilesetProperties(); break;
+    case Object::TileType:              addTileProperties(); break;
+    case Object::TerrainType:           addTerrainProperties(); break;
+    }
+
+    // Add a node for the custom properties
+    mCustomPropertiesGroup = mGroupManager->addProperty(
+            tr("Custom Properties"));
```

Comment:
I'd prefer to not wrap this line (I realize it's technically one character too long to fit in a terminal, but I had decided not to wrap it before).